robot-core/src/main/java/org/obolibrary/robot/QuotedEntityChecker.java

```diff
@@ -389,10 +389,15 @@ public OWLDataProperty getOWLDataProperty(@Nonnull String name) {
     if (iri != null) {
       return dataFactory.getOWLDataProperty(iri);
     }
-    if (ioHelper != null) {
-      iri = ioHelper.createIRI(name);
-      if (iri != null) {
-        return dataFactory.getOWLDataProperty(iri);
+    // prevent punning
+    if (!objectProperties.containsKey(name)) {
+      if (ioHelper != null) {
+        iri = ioHelper.createIRI(name);
+        if (iri != null) {
+          OWLDataProperty owlDataProperty = dataFactory.getOWLDataProperty(iri);
+          dataProperties.put(name, iri);
+          return owlDataProperty;
+        }
```

Comment:
Hm, can we link this to a ticket, not sure what this change is/means.

Reply:
Related with: https://github.com/owlcs/owlapi/issues/851
> Property punning between properties is illegal, but punning of a property name and class name or individual name is allowed